tests/ui/doc/doc_lazy_blockquote.rs

```diff
@@ -0,0 +1,47 @@
+#![warn(clippy::doc_lazy_continuation)]
+
+/// > blockquote with
+/// lazy continuation
+//~^ ERROR: doc quote missing `>` marker
+fn first() {}
+
+/// > blockquote with no
+/// > lazy continuation
+fn first_nowarn() {}
+
+/// > blockquote with no
+///
+/// lazy continuation
+fn two_nowarn() {}
+
+/// > nest here
+/// >
+/// > > nest here
+/// > lazy continuation
+//~^ ERROR: doc quote missing `>` marker
+fn two() {}
+
+/// > nest here
+/// >
+/// > > nest here
+/// lazy continuation
+//~^ ERROR: doc quote missing `>` marker
+fn three() {}
```

Comment:
I woud argue the applicability of these (and some other similar test cases below) being `MachineAppilcable`, maybe the `lazy continuation` was not a quote, but the user just forgot to add a newline before it, we would never know. Putting it in a nested quote automatically might cause confusion sometimes. 🤔